conda_build/build.py

```diff
@@ -1026,6 +1023,12 @@ def build(m, post=None, need_source_download=True, need_reparse_in_env=False, bu
         if excludes or output_excludes:
             exclude_pattern = re.compile('|'.join('(?:^{}(?:\s|$|\Z))'.format(exc)
                                             for exc in excludes | output_excludes))
+
+        # this metadata object may not be finalized - the outputs are, but this is the
+        #     top-level metadata.  We need to make sure that the variant pins are added in.
+        if m.meta.get('requirements'):
```

Comment:
I think `utils.insert_variant_versions(m.meta.get('requirements', {}), m.config.variant, 'build')` could be used to avoid the if block